src/lib/abbot/nostr.py

```diff
@@ -48,21 +46,34 @@ class AbbotNostr:
     notices = []
     events = []
 
-    def __init__(self, sk: str, filters_data: dict = None, author_whitelist: Optional[List[str]] = None):
-        self.private_key = PrivateKey.from_hex(sk)
-        self.public_key = self.private_key.public_key
-        self.filters_data = filters_data or DEFAULT_FILTERS
-        self.filters_list = FiltersList([Filters(**self.filters_data)])
+    def __init__(self, sk: str, custom_filters: Filters = None, author_whitelist: Optional[List[str]] = None):
+        self._private_key = PrivateKey.from_hex(sk)
+        self.public_key: PublicKey = self._private_key.public_key
+        self.author_whitelist: Optional[List[str]] = author_whitelist
         self.author_whitelist = author_whitelist
+        self.custom_filters = custom_filters
+        self.default_filters_list: List[Filters] = [
+            Filters(kinds=[DM, CHANNEL_CREATE, CHANNEL_MESSAGE], pubkey_refs=[BOT_NOSTR_PK], limit=1000),
+            Filters(kinds=[BOT_CHANNEL_INVITE], pubkey_refs=[BOT_NOSTR_PK], authors=self.author_whitelist, limit=1000),
+        ]
+
+    def _instantiate_direct_message(
+        self,
+        partner_pk: str,
+        cleartext: str | None,
+        encrypted: str | None,
+        reference_event_id: str | None = None,
```

Comment:
> there is a `typings.Optional` that could be used here
> 
> ```python
> cleartext: Optional[str],
> encrypted: Optional[str],
> reference_event_id: Optional[str] = None,
> ```

The original Optional type hint PEP 484 added a shortcut, `typing.Optional`:
```python
from typing import Optional

cleartext: Optional[str]
```

But since PEP 604, it’s recommended to use the union syntax:
```python
cleartext: str | None
```

The syntax might even become `str?` in the future: https://mobile.twitter.com/hynek/status/1581515882077237249

### TL;DR

Older syntax: `cleartext: Optional[str]`

Newer syntax: `cleartext: str | None`

But in the end, we can choose whichever we want, there's not really a "right" answer.